traitsui/editors/shell_editor.py

```diff
@@ -50,16 +50,21 @@ def init(self, parent):
         value = self.value
         if self.factory.share and isinstance(value, dict):
             locals = value
-        self._shell = shell = PythonShell(parent, locals=locals)
+        self._shell = shell = PythonShell(parent)
         self.control = shell.control
+        shell.interpreter().locals = locals
         if locals is None:
             object = self.object
             shell.bind("self", object)
             shell.observe(
                 self.update_object, "command_executed", dispatch="ui"
             )
             if not isinstance(value, dict):
-                object.observe(self.update_any, dispatch="ui")
+                object.observe(
+                    self.update_any,
+                    match(lambda name, ctrait: True),
```

Comment:
this wasn't causing the error, but it was a problem.  Not passing in a trait name in `on_trait_change` listened to all traits on the object.  This is the equivalent in observe

Reply:
i'm not sure how this was working earlier. what happens to `observe` when we don't pass in a trait name?